backend/app/Http/Controllers/ScoreController.php

```diff
@@ -36,10 +40,22 @@ public function calculateScore(Request $request)
         $isPassed = $result > $passingScore ? true : false;
 
         if($answers > 0){
+            $takenCategory = new TakenCategory();
+            $takenCategory->user_id = Auth::id();
+            $takenCategory->category_id = $categoryId;
+            $takenCategory->taken = 1;
+            $takenCategory->save();
```

Comment:
Much better if you use this
```suggestion
            new TakenCategory::create({
                "user_id" => Auth::id(),
                "category_id" => $categoryId,
                ...
            });
```